src/esargent.java

```diff
@@ -0,0 +1,129 @@
+			String [] multDiv = element.split("(?=[*])|(?<=[*])");   //Split into array including values, multiplication, and division symbols
+			long multResult = 0;
+			if (multDiv[0].equals("+") || multDiv[0].equals("-")) {
+				continue;
+			} else {
+				multResult = Long.parseLong(multDiv[0]);   //Initialize mult/div result as 1st element in the multiplication/division array
+			}
+			for (int i = 1; i < multDiv.length; i +=2) {   //Iterate throuh mult/div array by two to evaluate based on operand
+				long val = Long.parseLong(multDiv[i + 1]);
+				multResult = calc.multiply(multResult, val);
+			}
+			result = calc.add(result, multResult);
+		}
+		return result;
+	}
+	
+	public void doubleResult(String userInput) {
+		esargent es = new esargent();
+		//Perform calculations inside parentheses
+		while (userInput.contains("(")) {
+			String insidePar = userInput.substring(userInput.indexOf("("), (userInput.indexOf(")") + 1));
+			String insidePar2 = insidePar.replace("(", "");   //Create new string w/o parentheses
+			insidePar2 = insidePar2.replace(")", "");
+				
+			String[] plusMinus = insidePar2.split("(?=[+])|(?<=[+])");   //Split into array including values and addition symbols			
+			userInput = userInput.replace(insidePar, Double.toString(calcDouble(plusMinus)));   //Replace parentheses in original string with result				
+		}
+		//Perform calculations outside parentheses
+		String[] plusMinus2 = userInput.split("(?=[+])|(?<=[+])");   //Split into array including values and addition symbols
+		try {
+			if (Double.toString(es.calcDouble(plusMinus2)).equals("Infinity")) {   //Error message printed if user tries to divide by zero
+				System.out.println("Not Computable\n");
+			} else {
+				System.out.println(es.calcDouble(plusMinus2) + "\n");
+			}
+		} catch (NumberFormatException e) {
+			System.out.println("Not Computable\n");
+		}
+	}
+	
+	public void longResult(String userInput) {
+		esargent es = new esargent();
+		//Perform calculations inside parentheses
+		while (userInput.contains("(")) {
+			String insidePar = userInput.substring(userInput.indexOf("("), (userInput.indexOf(")") + 1));
+			String insidePar2 = insidePar.replace("(", "");   //Create new string w/o parentheses
+			insidePar2 = insidePar2.replace(")", "");
+			
+			String[] plusMinus = insidePar2.split("(?=[+])|(?<=[+])");   //Split into array including values and addition symbols		
+			userInput = userInput.replace(insidePar, Long.toString(calcLong(plusMinus)));   //Replace parentheses in original string with result				
+		}
+		//Perform calculations outside parentheses
+		String[] plusMinus2 = userInput.split("(?=[+])|(?<=[+])");   //Split into array including values and addition symbols	
+		//System.out.println(calcLong(plusMinus2) + "\n");
+		try {
+			System.out.println(es.calcLong(plusMinus2) + "\n");
+		} catch (NumberFormatException e) {
+			System.out.println("Not Computable\n");
+		}
+	}
+	
+	public static void main(String[] args) {
```

Comment:
Typically, you would not want to start in a main statement for the class. Your better options would be to derive the System.in from the main class and pass that into the constructor for your class and then operate directly on the class.